src/InverseKinematics.cpp

```diff
@@ -0,0 +1,154 @@
+#include "InverseKinematics.h"
+
+
+InverseKinematics::InverseKinematics()
+:updated(false)
+,initialized(false)
+,alreadyOptimized(false)
+{
+    solverPointer = new InverseKinematicsIPOPT();
+    loader = IpoptApplicationFactory();
```

Comment:
Than I think you have a cyclic reference (I'm not really sure is the correct term) which is something to be avoided when using ``Ipopt::SmartPtr`` according to their documentation (if I have understood well).